services/QuillLMS/engines/evidence/lib/evidence/evidence/plagiarism_check.rb

```diff
@@ -124,13 +124,20 @@ def feedback_object
     end
 
     private def identify_matched_slices(entry_slices, passage_slices)
-      entry_slices.select do |_, slice|
-        # Check to see if there's enough similarity for it to be worth doing a Levenshtein comparison
+      contiguous_matched_indexes = []
+      # Placeholder to be overridden during the first run of the loop that makes it past the confirm_minimum_overlap? guard statement.  If that never happens, we don't have to calculate this value
+      passage_slice_strings = nil
+      entry_slices.each do |index, slice|
```

Comment:
I wonder if some of this code can be simplified, since we've simplified the logic requirements.

So the logic was previously to `"find all of the matching indexes, then find the longest set of consecutive indexes, form the string from them"`, 

Now we are doing `"Find all of the matching indexes, then stop once they aren't consecutive, then find the longest set of consecutive indexes, form the string from them"`. There's a bit of redundancy there.

Reply:
Good catch.  I've updated that code so that the main Levenshtein loop just assembles the string, rather than a series of indexes that gets combined into a string later.  Benchmarks don't show this speeding things up very much, but it's probably much more readable than it was.